src/ast.rs

```diff
@@ -421,19 +426,33 @@ pub struct FuncCall {
 
 impl FuncCall {
     pub fn new_with_casts(
-        cn: Option<CascadeString>,
+        cn: Option<(CascadeString, Option<CascadeString>)>,
         n: CascadeString,
         a: Vec<(Argument, Option<CascadeString>)>,
     ) -> FuncCall {
+        if let Some(some_cn) = cn {
```

Comment:
This "if let" followed by handling for the None case really should be a match block.